buildSrc/src/test/java/org/elasticsearch/gradle/test/GradleUnitTestCase.java

```diff
@@ -0,0 +1,14 @@
+package org.elasticsearch.gradle.test;
+
+import com.carrotsearch.randomizedtesting.JUnit4MethodProvider;
+import com.carrotsearch.randomizedtesting.RandomizedRunner;
+import com.carrotsearch.randomizedtesting.annotations.TestMethodProviders;
+import org.junit.runner.RunWith;
+
+@RunWith(RandomizedRunner.class)
+@TestMethodProviders({
+    JUnit4MethodProvider.class,
+    JUnit3MethodProvider.class
+})
+public abstract class GradleUnitTestCase extends BaseTestCase  {
```

Comment:
Why have this and BaseTestCase? I think just having one of these would be fine?

Reply:
An integration test is not a unit test, so I didn't initially want them to inherit from each-other, but would have to extend the name checker to be able to deal with it.